### ISSUE TITLE
[OC-8.0 update] defaultapp => calendar leads to infinite redirect

### ISSUE THREAD

**Issue by RandolfCarter** [owncloud/core#14225]
### Steps to reproduce
1. Do manual upgrade to OC8.0 as described here: http://doc.owncloud.org/server/7.0/admin_manual/maintenance/upgrade.html
2. Go to login screen
3. Enter credentials, press Login

### Expected behaviour
Get logged in and see default app.

### Actual behaviour
Since calendar app wasn't shipped, owncloud can't navigate to it. That's fine of course, but it shouldn't result in a redirect loop. A nice error message like "The app 'calendar' isn't available" or something along those lines would be great!

Apart from this easily fixable thing, I like OC8 quite a lot! Keep up the good work!

### Server configuration
**Operating system**: Ubuntu Server 14.04.1 LTS
**Web server:** Apache 2.4.9
**Database:** mysql  Ver 15.1 Distrib 10.0.16-MariaDB
**PHP version:** 5.5.9-1ubuntu4.5
**ownCloud version:** Upgraded from 7.04 to 8.0
**Updated from an older ownCloud or fresh install:** installation started with 4.0
**List of activated apps:** (in addition to standard)
* mail
* music
* news
* notes
* shorty
* tasks

**The content of config/config.php:**
```
<?php
define('DEBUG',true);
$CONFIG = array (
  'datadirectory' => '/media/ownclouddata',
  'dbtype' => 'mysql',
  'version' => '8.0.0.7',
  'dbname' => 'owncloud',
  'dbhost' => '127.0.0.1',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => '<dbpassword>',
  'installed' => true,
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'localhost',
  'mail_smtpport' => '587',
  'mail_smtpauth' => true,
  'mail_smtpname' => 'smtpuser',
  'mail_smtppassword' => 'smtppass',
  'mail_smtpsecure' => 'tls',
  'maxZipInputSize' => 1073741824,
  'allowZipDownload' => true,
  'loglevel' => '1',
  'email' => 'owncloud@mydomain',
  'instanceid' => '<instanceid>',
  'secretword' => '<secretword>',
  'siteemail' => 'sharedgallery@mydomain',
  'defaultapp' => 'files',
  'maintenance' => false,
  'forcessl' => true,
  'passwordsalt' => '<passwordsalt>',
  'knowledgebaseenabled' => false,
  'trusted_domains' =>  
  array (
    0 => 'mydomain',
    1 => 'mydomain',
    2 => 'lanserverhostname',
  ),  
  'theme' => '', 
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'preview_max_scale_factor' => 1,
  'mail_from_address' => 'owncloud',
  'mail_domain' => 'mydomain',
  'appstoreenabled' => true,
  'overwritewebroot' => '/owncloud',
#  'appcodechecker' => 'false',
  'secret' => '<secret>',
);
```
not sure why the double "mydomain" entry is there, I suppose I entered it once with port number (as I use a non-standard port, and it got removed during upgrade or something?

**Are you using external storage, if yes which one:** local
**Are you using encryption:** no

### Client configuration
**Browser:** Firefox 35.0.1
**Operating system:** Ubuntu 14.10

#### Web server error log
```
[Sat Feb 14 16:16:17.616232 2015] [:error] [pid 7089] [client localip:33247] PHP Fatal error:  require_once(): Failed opening required '/appinfo/remote.php' (include_path='/var/www/owncloud/3rdparty/phpseclib/phpseclib/phpseclib:/var/www/owncloud/3rdparty/pear/pear_exception:/var/www/owncloud/3rdparty/pear/archive_tar:/var/www/owncloud/lib/private:/var/www/owncloud/config:/var/www/owncloud/3rdparty:/var/www/owncloud/apps:/var/www/owncloud/lib:.:/usr/share/php:/usr/share/pear:/var/www/owncloud') in /var/www/owncloud/remote.php on line 54
```

#### ownCloud log (data/owncloud.log)
basically the same as in webserver log

#### Browser log
not relevant

**Comment by PVince81** [comment]
Confirmed. Had a quick look to find a way to fix it but it seems to be not obvious.
One idea would be to add a check here https://github.com/owncloud/core/blob/master/lib/base.php#L821 (or in that function) to see if the URL we redirect to is the same as the current URL.
If yes, then redirect to the "files" app instead.

Not sure how to rebuild the full URL from the various pieces of OC_Request. @LukasReschke any idea ?

**Comment by DeepDiver1975** [comment]
Well - the root cause is that an out dated calendar app is still enabled.
In case the app would have been disabled on upgrade this problem does not exist.

refs #14026 - which is what I'm currently working on

**Comment by RandolfCarter** [comment]
great to hear! didn't think of disabling calendar since it used to be a shipped app ;).
so this can be closed I guess...

**Comment by HLFH** [comment]
Same issue on OC8. Any updates?
